<?xml version="1.0"?>
<launch>
<!--argument-->
    <arg name="points_topic" default="/lidar/vlp32_middle/PointCloud2" />
    <arg name="map_tf" default="odom" />
    <arg name="base_lidar_tf" default="base_foot" />
<!--    <arg name="map_tf" default="map" />-->
<!--    <arg name="base_lidar_tf" default="velodyne" />-->
    <arg name="downsample_resolution" default="0.1" />
    <arg name="init_x" default="0.0" />
    <arg name="init_y" default="0.0" />
    <arg name="init_yaw" default="0.0" />
    <arg name="trim_low" default="-1.8" />
    <arg name="trim_high" default="3" />
    <arg name="radius" default="25.0" />
    <arg name="mapUpdateTime" default="8" />
    <arg name="global_map_pcd_path" default="/home/vickylzy/WorkSPacesROS/catkin_ws/src/prm_localization/data/XJTU_underground_parkinglot.pcd" />
<!--icp para-->
    <arg name="icpMaximumIterations" default="15" />
    <arg name="icpRANSACOutlierRejectionThreshold" default="0.08" />
    <arg name="icpTransformationEpsilon" default="1e-5" />
<!--    filter-->
    <arg name="farPointThreshold" default="20" />
    <arg name="nearPointThreshold" default="1.3" />


<!--manager-->
    <node pkg="nodelet" type="nodelet" name="manager" args="manager" output="screen"/>
<!--nodelet-->
<!-- same meaning param filed in two nodes must have same values-->
    <node pkg="nodelet" type="nodelet" name="rt_localization_nodelet" args="load prm_localization/RealTime_Localization manager">
        <remap from="/velodyne_points" to="$(arg points_topic)" />
        <param name="farPointThreshold" value="$(arg farPointThreshold)" />
        <param name="nearPointThreshold" value="$(arg nearPointThreshold)" />
        <param name="trim_low" value="$(arg trim_low)" />
        <param name="trim_high" value="$(arg trim_high)" />
        <param name="init_x" value="$(arg init_x)" />
        <param name="init_y" value="$(arg init_y)" />
        <param name="init_yaw" value="$(arg init_yaw)" />
        <param name="icpMaximumIterations" value="$(arg icpMaximumIterations)" />
        <param name="icpRANSACOutlierRejectionThreshold" value="$(arg icpRANSACOutlierRejectionThreshold)" />
        <param name="icpTransformationEpsilon" value="$(arg icpTransformationEpsilon)" />
        <param name="map_tf" value="$(arg map_tf)" />
        <param name="base_lidar_tf" value="$(arg base_lidar_tf)" />
        <param name="downSampleSize" value="$(arg downsample_resolution)"  />
    </node>
    <node pkg="nodelet" type="nodelet" name="globalmap_provider_nodelet" args="load prm_localization/GlobalmapProviderNodelet manager">
        <param name="downsample_resolution" value="$(arg downsample_resolution)" />
        <param name="map_tf" value="$(arg map_tf)" />
        <param name="base_lidar_tf" value="$(arg base_lidar_tf)" />
        <param name="init_x" value="$(arg init_x)" />
        <param name="init_y" value="$(arg init_y)" />
        <param name="init_yaw" value="$(arg init_yaw)" />
        <param name="radius" value="$(arg radius)" />
        <param name="mapUpdateTime" value="$(arg mapUpdateTime)" />
        <param name="global_map_pcd_path" value="$(arg global_map_pcd_path)" />
    </node>

</launch>
<!--        output="screen"-->
<!--    <node pkg="nodelet" type="nodelet" name="prm_localization_nodelet" args="load prm_localization/PRMLocalizationNodelet manager"/>-->
